### PR TITLE
update userMismatch check

### PR DESCRIPTION
**Pull request by weijjia**
only return user mismatch when user is passed in the request and it doesn't match either displayable and unique userid.

### CODE REVIEW

**Comment by weijjia** [comment]
@RandalliLama @addev-ashish

File: src/src/com/microsoft/aad/adal/TokenCacheAccessor.java [hunk 1]
```diff
@@ -273,18 +273,14 @@ private void setItemToCacheForUser(final String resource, final String clientId,
     }
     
     private boolean isUserMisMatch(final String user, final TokenCacheItem tokenCacheItem) {
-        // when both user and userInfo in AuthenticationResult is null, non-mismatch case
-        if (StringExtensions.IsNullOrBlank(user) && tokenCacheItem.getUserInfo() == null) {
-            return false;
+        // If user if provided, it needs to match either displayId or userId. 
+        if (!StringExtensions.IsNullOrBlank(user)) {
+            return !user.equalsIgnoreCase(tokenCacheItem.getUserInfo().getDisplayableId())
```
Comment (addev-ashish) [inline src/src/com/microsoft/aad/adal/TokenCacheAccessor.java]: tokenCacheItem.getUserInfo was checked for null earlier. you should add it back

File: src/src/com/microsoft/aad/adal/TokenCacheAccessor.java [hunk 2]
```diff
@@ -273,18 +273,14 @@ private void setItemToCacheForUser(final String resource, final String clientId,
...
-        // If either user or userInfo in AuthenticationResult is null, there is a mismatch
-        if (StringExtensions.IsNullOrBlank(user) || tokenCacheItem.getUserInfo() == null) {
-            return true;
-        }
-        
-        return !user.equalsIgnoreCase(tokenCacheItem.getUserInfo().getDisplayableId()) 
-                && !user.equalsIgnoreCase(tokenCacheItem.getUserInfo().getUserId());
+        return false;
+
     }
```
Comment (RandalliLama) [inline src/src/com/microsoft/aad/adal/TokenCacheAccessor.java]: All those '!'s make my eyes bug out.  Would you cringe at:

```    
    final boolean isValidUser = !StringExtensions.IsNullOrBlank(user);
    if (isValidUser) {
        final boolean userMatchesDisplayableId
            = user.equalsIgnoreCase(tokenCacheItem.getUserInfo().getDisplayableId());

        final boolean userMatchesUserId
            = user.equalsIgnoreCase(tokenCacheItem.getUserInfo().getUserId());

        return !(userMatchesUserId || userMatchesDisplayableId); 
    } 
```

**Comment by addev-ashish** [comment]
:shipit:

**Pull request merged**